redis/src/main/java/org/togglz/redis/RedisStateRepository.java

```diff
@@ -0,0 +1,171 @@
+package org.togglz.redis;
+
+import org.togglz.core.Feature;
+import org.togglz.core.repository.FeatureState;
+import org.togglz.core.repository.StateRepository;
+import org.togglz.core.repository.util.DefaultMapSerializer;
+import org.togglz.core.repository.util.MapSerializer;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.Protocol;
+
+import java.util.Map;
+
+/**
+ * A state repository which stores the feature state in Redis.
+ * <p>
+ * The class provides a builder which can be used to configure the state repository instance
+ * (e.g. Redis hostname, Redis/Jedis configuration, serialization).
+ * </p>
+ *
+ * @author Cosmin Rentea
+ */
+public class RedisStateRepository implements StateRepository {
+
+    public static final String KEY_PREFIX = "togglz-";
+    public static final String ENABLED_FIELD = "enabled";
+    public static final String STRATEGY_FIELD = "strategy";
+    public static final String PARAMETERS_FIELD = "parameters";
+
+    protected final JedisPool jedisPool;
+    protected final JedisPoolConfig jedisPoolConfig;
+    protected final String hostname;
+    protected final String keyPrefix;
+    protected final MapSerializer mapSerializer;
+
+    public RedisStateRepository() {
+        jedisPoolConfig = null;
+        hostname = Protocol.DEFAULT_HOST;
+        keyPrefix = KEY_PREFIX;
+        mapSerializer = DefaultMapSerializer.multiline();
+        jedisPool = createJedisPool();
+    }
+
+    private RedisStateRepository(final Builder builder) {
+        jedisPoolConfig = builder.jedisPoolConfig;
+        hostname = builder.hostname;
+        keyPrefix = builder.keyPrefix;
+        mapSerializer = builder.mapSerializer;
+        jedisPool = createJedisPool();
+    }
+
+    private JedisPool createJedisPool() {
+        return jedisPoolConfig != null ?
+                new JedisPool(jedisPoolConfig, hostname) :
+                new JedisPool();
+    }
+
+    @Override
+    public FeatureState getFeatureState(final Feature feature) {
+        try (Jedis jedis = jedisPool.getResource()) {
+            final Map<String, String> map = jedis.hgetAll(keyPrefix + feature.name());
+            if (map == null || map.size() == 0) {
+                return null;
+            }
+            final FeatureState featureState = new FeatureState(feature);
+            featureState.setEnabled(Boolean.valueOf(map.get(ENABLED_FIELD)));
+            featureState.setStrategyId(map.get(STRATEGY_FIELD));
+            final String strategyParameters = map.get(PARAMETERS_FIELD);
+            if (strategyParameters != null) {
+                for (final Map.Entry<String, String> entry : mapSerializer.deserialize(strategyParameters).entrySet()) {
```

Comment:
Would you like to prefix all parameters with some string constant, like: "parameters-" , and store/retrieve them individually (so that no naming collision can occur) ?